[VPlan][NFC] Add new getMiddleBlock interface to VPlan

david-arm opened the pull request:
This work is in preparation for PRs #112138 and #88385 where
the middle block is not guaranteed to be the immediate successor
to the region block. I've simply add new getMiddleBlock()
interfaces to VPlan that for now just return

cast<VPBasicBlock>(VectorRegion->getSingleSuccessor())

Once PR #112138 lands we'll need to do more work to discover
the middle block.

fhahn commented on llvm/lib/Transforms/Vectorize/VPlan.h:
```diff
@@ -3689,6 +3689,14 @@ class VPlan {
   VPBasicBlock *getEntry() { return Entry; }
   const VPBasicBlock *getEntry() const { return Entry; }
 
+  /// Methods to support access to the middle block.
```
Might be good to add a bit more context, e.g. something like

```suggestion
  /// Returns the 'middle' block of the plan, that is the block that selects whether to execute the scalar tail loop or the exit block from the loop latch.
```

(or something nicer worded)

fhahn reviewed (commented):
Thanks for cleaning this up!

fhahn reviewed (approved):
LGTM, thanks!

david-arm committed:
[VPlan][NFC] Add new getMiddleBlock interface to VPlan

This work is in preparation for PRs #112138 and #88385 where
the middle block is not guaranteed to be the immediate successor
to the region block. I've simply add new getMiddleBlock()
interfaces to VPlan that for now just return

cast<VPBasicBlock>(VectorRegion->getSingleSuccessor())

Once PR #112138 lands we'll need to do more work to discover
the middle block.

david-arm commented:
Rebase and address conflicts
